core-services/src/main/java/org/zalando/nakadi/service/publishing/EventPublisher.java

```diff
@@ -201,7 +197,8 @@ private void partition(final List<BatchItem> batch, final EventType eventType)
         for (final BatchItem item : batch) {
             item.setStep(EventPublishingStep.PARTITIONING);
             try {
-                final String partitionId = partitionResolver.resolvePartition(eventType, item.getEvent());
+                final String partitionId =
```

Comment:
This change is going to break partitioning for log-compacted event types, it seems.  Previously the partition was calculated based on `partition_key_fields` and now it is using `partition_compaction_key` (due to the changes in `setEventKey` function below)...

At the same time, all tests passed :(

Reply:
I think in this situation we need the following two attributes instead of one:
```java
String compactionKey;
List<String> partitioningKeys;
```
Either (or both) may be null and both can be set at the same time.

It's a pity of course that we have distinct keys for partitioning and compaction, but it's too late to fix that.

Reply:
> This change is going to break partitioning for log-compacted event types, it seems

it still takes event key for log compacted event types from `partition_compaction_key`, but for hash partitioning it uses event values from partition keys